src/main/java/org/folio/rest/impl/PutOrdersByIdHelper.java

```diff
@@ -123,26 +130,40 @@ private boolean isPoNumberChanged(JsonObject poFromStorage, CompositePurchaseOrd
   }
 
   private CompletableFuture<Void> updatePoLines(JsonObject poFromStorage, CompositePurchaseOrder compPO) {
-    if (isNotEmpty(compPO.getCompositePoLines()) || isPoNumberChanged(poFromStorage, compPO)) {
-      return getPoLines(poFromStorage.getString(ID), lang, httpClient, ctx, okapiHeaders, logger)
-        .thenCompose(jsonObject -> {
-          JsonArray existedPoLinesArray = jsonObject.getJsonArray(PO_LINES);
-          if (isNotEmpty(compPO.getCompositePoLines())) {
-            return handlePoLines(compPO, existedPoLinesArray);
-          } else {
-            return updatePoLinesNumber(compPO, existedPoLinesArray);
-          }
-        });
-    } else {
-      return completedFuture(null);
-    }
+    return getPoLines(poFromStorage.getString(ID), lang, httpClient, ctx, okapiHeaders, logger)
+      .thenCompose(polinesFromStorage -> {
+        JsonArray existedPoLinesArray = polinesFromStorage.getJsonArray(PO_LINES);
+        if (isNotEmpty(compPO.getCompositePoLines())) {
+          return handlePoLines(compPO, poFromStorage, existedPoLinesArray);
+        } else if (isPoNumberChanged(poFromStorage, compPO)) {
+          return updatePoLinesNumber(compPO, existedPoLinesArray);
+        } else if (isEmpty(compPO.getCompositePoLines()) && (existedPoLinesArray != null)) {
```

Comment:
This check is not required because of internal agreement. Please refer to [order.raml](https://github.com/folio-org/mod-orders/blob/master/ramls/order.raml#L69):
>          Update a purchase order with given {id}
>          - if request does not include po_lines or includes "po_lines": [] or "po_lines": null - update just purchase order summary
>          - if request includes array of "po_lines" - update PO lines as per request

Reply:
implementation fixed